src/store/Reducers/SystemStore.js

```diff
@@ -161,43 +34,14 @@ const deselectRow = (state, id) => (
     deselectRowsByIds(state, [id])
 );
 
-export const systemsReducer = (INVENTORY_ACTION, columns) => applyReducerHash({
-    ['GET_SYSTEMS_PENDING']: (state) => ({
-        ...state,
-        rows: [],
-        systems: undefined,
-        systemsCount: undefined,
-        columns,
-        loaded: false
-    }),
-    ['GET_SYSTEMS_FULFILLED']: (state, { systems, systemsCount }) => ({
-        ...state,
-        systems,
-        systemsCount,
-        total: systemsCount,
-        rows: systemsToRows(systems).map((row) => ({
-            ...row, selected: isSelected(row.id, state.selectedEntities)
-        })),
-        columns,
-        loaded: true
-    }),
+export const entitiesReducer = (INVENTORY_ACTION, columns) => applyReducerHash({
     [INVENTORY_ACTION.LOAD_ENTITIES_PENDING]: (state) => ({
         ...state,
-        total: state.systemsCount,
-        rows: state.systems !== undefined ? systemsToRows(state.systems).map((row) => ({
-            ...row, selected: isSelected(row.id, state.selectedEntities)
-        })) : [],
-        columns,
-        loaded: state.systemsCount !== undefined
+        columns
     }),
     [INVENTORY_ACTION.LOAD_ENTITIES_FULFILLED]: (state) => ({
         ...state,
-        total: state.systemsCount,
-        rows: state.systems !== undefined ? systemsToRows(state.systems).map((row) => ({
-            ...row, selected: isSelected(row.id, state.selectedEntities)
-        })) : [],
-        columns,
-        loaded: state.systemsCount !== undefined
+        columns: state.total > 0 ? columns : [{ title: '' }]
     }),
     [SELECT_ENTITY]: (state, { payload: { id, selected, clearAll } }) => {
```

Comment:
We should be able to remove this from our constants and use:

```suggestion
    [INVENTORY_ACTION.SELECT_ENTITY]: (state, { payload: { id, selected, clearAll } }) => {
```

Reply:
Why do we even need to keep this reducer? If there's extra functionality that would be useful for the inventory table to provide, we should add it there. I agree `SELECT_ENTITY` is a bit cumbersome to use without a select all/none, so let's contribute this back.

Reply:
Unfortunately 'SELECT_ENTITY' isn't exported[0] as far as I can see. :/

> Why do we even need to keep this reducer?

We don't. I was close to ripping it out. (I had a facepalm moment this morning and @vkrizan too) We don't need to interact or use the inventory store at all to keep selections, even now (in master) we could easily put this all into a simple `useState`. The more complex solution would be `useBulkSelect`[1]. I'm not sure anymore, I believe we started doing this with the `selectedEntities` because there was no other way to mark rows as selected (maybe). But we should definitely get rid of this. 

https://github.com/RedHatInsights/frontend-components/blob/39ab58e14990887dff0f2c42ef0c5a20f0f3e11b/packages/inventory/src/redux/action-types.js#L29

Reply:
That's interesting about the event not being exposed... It is documented here https://github.com/RedHatInsights/frontend-components/blob/master/packages/inventory/doc/inventory.md#inventory-actions. How would we get the checkbox event out of the Inventory w/o listening on that event (state change)?

Reply:
I really like `useSelector`. It's really clean and doesn't clutter our code with redux stuff.

Reply:
If SELECT_ENTITY isn't exported but it'd be useful to be exported, then let's go export it! :wink: Unless the platform UI folks have a reason against it, we can claim we need some functionality not provided. Maybe they'd have an even more elegant solution for us.

Reply:
Wait... it is exported, just not in INVENTORY_ACTIONS: https://github.com/RedHatInsights/frontend-components/blob/39ab58e14990887dff0f2c42ef0c5a20f0f3e11b/packages/inventory/src/redux/action-types.js#L35

Reply:
@vkrizan That is interesting! Either the docs are not up to date or the code. :D

> How would we get the checkbox event out of the Inventory w/o listening on that event (state change)?

We should be able to use the plain Patternfly table `onSelect` prop. There is might even an FECInventoryTable specific prop.

Reply:
@akofink We also would need to add the whole inventory package just to get the SELECT_ENTITY constant, i think. 

Again, we can and probably eventually will remove that part. It works for now.